report/metrics.go

```diff
@@ -159,33 +145,36 @@ func (m Metric) Add(t time.Time, v float64) Metric {
 
 // Merge combines the two Metrics and returns a new result.
 func (m Metric) Merge(other Metric) Metric {
-	// Merge two lists of samples in O(n)
-	curr1, curr2, acc := m.Samples, other.Samples, make([]interface{}, 0, m.Len()+other.Len())
-	var newSamples ps.List
+	// Merge two lists of Samples in O(n)
 
+	// TODO: be smarter and check for non-overlapping metrics with first and last?
```

Comment:
That seems reasonable. Metrics will be non-overlapping most of the time, won't they?